src/main/scala/amailp/intellij/robot/psi/KeywordDefinition.scala

```diff
@@ -45,4 +48,27 @@ object KeywordDefinition {
   }
 
   def findInFile(file: RobotPsiFile) = PsiTreeUtil.findChildrenOfType(file.getNode.getPsi, classOf[KeywordDefinition]).toSet
-}
\ No newline at end of file
+
+  def findMatchingInLibraries(libraries: Iterable[Library], project: Project, reference: String) = findInPythonFiles(libraries, project, reference) ++
+      findInPythonClasses(libraries, project, reference)
+
+  def findInPythonFiles(libraries: Iterable[Library], project: Project, reference: String) = {
+    for {
+      library <- libraries
+      pyFile <- PyModuleNameIndex.find(library.getText, project, true)
+      keyword <- findInPythonFile(pyFile)
+      if (keyword.getName.toLowerCase matches reference.replaceAll(" ", "_").toLowerCase) && Option(keyword.getContainingClass).isEmpty
```

Comment:
please extract a method and remove duplication for:
keyword.getName.toLowerCase matches reference.replaceAll(" ", "_").toLowerCase